pkg/resource/resource.go

```diff
@@ -212,6 +214,11 @@ func IsAPIError(err error) bool {
 	return ok
 }
 
+// IsAPIErrorWrapped returns true if err is a K8s API error, or recursively wraps a K8s API error
+func IsAPIErrorWrapped(err error) bool {
```

Comment:
I do not want to change the behavior of existing resource.IsAPIError because I don't know all the contexts it's used.